### ISSUE TITLE
vue/require-toggle-inside-transition false negative

### ISSUE THREAD

**Issue by yoshi-pi** [vuejs/eslint-plugin-vue#2307]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.52.0
- **eslint-plugin-vue version:** 9.18.1
- **Node version:** v18.15.0
- **Operating System:** macOS 13.5.2

**Please show your full configuration:**

```json5
/* eslint-env node */
module.exports = {
  root: true,
  'extends': [
    'plugin:vue/vue3-essential',
    'eslint:recommended'
  ],
  parserOptions: {
    ecmaVersion: 'latest'
  }
}
```

**What did you do?**	

```vue
<template>
  <Transition>
    <FooComp /> // should throw error IMO
  </Transition>
  <Transition>
    <div v-if="true">A</div>
    <div>B</div> // should throw error IMO
  </Transition>
</template>

```

**What did you expect to happen?**
`<FooComp />` and `<div>B</div>` should throw errors because of [#2290](https://github.com/vuejs/eslint-plugin-vue/issues/2290#issuecomment-1776864807).

**What actually happened?**

There are no errors.

**Repository to reproduce this issue**

https://github.com/yoshi-pi/bug-report2-require-toggle-inside-transition

**Comment by ota-meshi** [comment]
Thank you for posting the issue.

But I don't think they should be reported in `vue/require-toggle-inside-transition` rule.

```vue
<template>
  <Transition>
    <FooComp />
  </Transition>
</template>
```

If FooComp can switch v-if internally it works fine.
I think if you never use v-if in your component and want it to be reported in `vue/require-toggle-inside-transition`, we need to add the option.
Could you please open the issue and PR suggesting options if you want it?

```vue
<template>
  <Transition>
    <div v-if="true">A</div>
    <div>B</div>
  </Transition>
</template>
```

This is an incorrect `<template>`, but I don't think it's something that should be checked by `vue/require-toggle-inside-transition` rule. If you want that checked, I think we need a new rule.
If you need a new rule, could you please open an issue to suggest a new rule and a PR to add the rule?
